### DISCUSSION TITLE
Animations

### DISCUSSION THREAD

**Discussion by lab596** [hdescottes/GdxGame#45] [General]
Hello I noticed that in your improve section you mentioned animations and also in bugs you mentioned a fade in issue. I am assuming that you are using JPanels and Frames that implement button action listeners to call upon these said animations. 

I am not able to find you code for where this occurs and if you could point out where it is I can take a look. 

Anyway other than the animations libraries for java I think a way to get a fade in effect is to use the graphics engine and simply paint a black frame that lowers in opacity giving it the fade animation. This would occur with a for loop that starts with 100 (max opacity) and the "i" in the for loop decreases which is set to the opacity of the graphics black screen. This may be too fast so a timer or a delay may be necessary. 

This is all assuming that you do not already have a fade in implemented.
Otherwise I would love to take a look at the code if you could direct me. 

Thanks.

**Comment by hdescottes** [comment]
Hello, first thank you for your interest in my project ! Every contributions for the bugs and graphics are welcome.
My repository is public so you can find it on my profile (but i can see that you forked it :) )

The animations, that i am referring to, are attacks animations (for now) during battles. The idea was to recreate some sprites and templates of the characters that made this possible. Currently there is only the walking animation available and as i am not very proud of my graphic skills... Maybe people could help me on this part !
There are also map animations like opening doors, water animations. Those, i think, will be possible with some additional layers on the maps.

Regarding the fade in issue, it is relating to the transition menu screen -> game screen that yoiu can see with the TODO annotation in class "MenuNewGameScreen" line 99 and 135 (as well in MenuLoadGameScreen line 95).
The fade in transition is a custom one. You can see them in the package "transition" under "screen".

Feel free to take a look and/or come with solutions or advices.

Thanks

**Reply by lab596** [reply]
Hi so I took a look at what you had and I think that you need a render method that will actually create the transition meaning that it will make the screen black. Im not sure if you have something like this but I think you may need code that looks like this: 
public void render(Screen current, Screen next) {
		current.render();
		color.set(0f, 0f, 0f, getAlpha());
		// draw a quad over the screen using the color
	}
 to actually draw the black color over the current panel. 
You will later need to set it to the next panel to remove the fade. I'm not sure if you have something similar but I think this website sums up how to do it pretty well : https://blog.gemserk.com/2012/03/05/implementing-transitions-between-screens/

**Reply by hdescottes** [reply]
Actually, i already have something like this : [FadeInTransitionEffect](https://github.com/hdescottes/GdxGame/blob/master/core/src/main/java/com/gdx/game/screen/transition/effects/FadeInTransitionEffect.java)
And as you can see, they directly inspired me for this implementation :)
This is working pretty well during any transition between screens, except from menu -> game and i don't know why.

If you try to implement this transition effect for the battle screen for example, it will work just fine.
`ArrayList<TransitionEffect> effects = new ArrayList<>();`
`effects.add(new FadeOutTransitionEffect(1f));`
`effects.add(new FadeInTransitionEffect(1f));`
`setScreenWithTransition((BaseScreen) gdxGame.getScreen(), new BattleScreen(game, playerHUD, mapManager, resourceManager), effects);`
In [GameScreen](https://github.com/hdescottes/GdxGame/blob/master/core/src/main/java/com/gdx/game/screen/GameScreen.java) line 186
